deployer/src/handlers/error.rs

```diff
@@ -0,0 +1,42 @@
+use std::error::Error as StdError;
+
+use axum::http::{header, HeaderValue, StatusCode};
+use axum::response::{IntoResponse, Response};
+use axum::Json;
+
+use serde::{ser::SerializeMap, Serialize};
+use serde_json::json;
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error("Persistence failure: {0}")]
+    Persistence(#[from] crate::persistence::PersistenceError),
+}
+
+impl Serialize for Error {
+    fn serialize<S>(&self, serializer: S) -> std::result::Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        let mut map = serializer.serialize_map(Some(2))?;
+        map.serialize_entry("type", &format!("{:?}", self))?;
+        map.serialize_entry("msg", &self.source().unwrap().to_string())?;
```

Comment:
As per `thiserror` doc `.source().unwrap()` is safe only for variants that have `#[from]` or `#[source]` so we should be a bit careful - perhaps we can just `.unwrap_or_default` or something?

Reply:
I agree, this is a good point. Since we implement `Display` through `thiserror` for all the deployer errors, we can do `&self.source().unwrap_or(self).to_string()`. This way we will get either the source or the `Display` representation. If we add a new error to `handlers::Error`, this will not compile if the new error doesn't implement `Display`.